install.sh

```diff
@@ -64,26 +64,39 @@ function clone_repo() {
 }
 
 function unjoin() {
+  for id in $(kubectl get foreignclusters -o jsonpath="{.items[*].metadata.name}"); do
+    kubectl patch foreignclusters "$id" -p '{"spec":{"join":false}}' --type 'merge'
+  done
+
+  for id in $(kubectl get advertisements -o jsonpath="{.items[*].metadata.name}"); do
+    kubectl patch advertisements "$id" -p '{"status":{"advertisementStatus":"Deleting"}}' --type 'merge'
+  done
+
   kubectl patch clusterconfig configuration -p '{"spec":{"advertisementConfig":{"outgoingConfig":{"enableBroadcaster":false}}}}' --type 'merge'
 
   retry=10
-  while [ "$(kubectl get pods -n "$NAMESPACE" | grep -c broadcaster)" -gt 0 ]; do
+  while [ "$(kubectl get pods -n "$NAMESPACE" | grep -c 'broadcaster')" -gt 0 ] || [ "$(kubectl get pods -n "$NAMESPACE" | grep -c 'virtual-kubelet')" -gt 0 ]; do
     sleep 10
     retry=$((retry-1))
     if [ "$retry" -eq 0 ]; then
       echo "Max retries reached: unable to unpeer all clusters"
       exit 1
     fi
   done
-
-  for id in $(kubectl get foreignclusters -o jsonpath="{.items[*].metadata.name}"); do
-    kubectl patch foreignclusters "$id" -p '{"spec":{"join":false}}' --type 'merge'
-  done
 }
 
 function deleteCrd() {
   echo "delete CRD"
   clone_repo
+
+  # there are only resources that are been replicated during unjoin process, finalizers are useless
```

Comment:
AFAIK, replicated resources should not have finalizers. Correct @alacuku ?

Reply:
A replicated resource in a remote cluster may have finalizers. For example, if an operator that processes a replicated resource need to do some clean-up after the deletion of the resource than it may set the finalizers for this purpose.
If your question is if we replicate also the finalizers then the answer is: no we do not.